### PR TITLE
Implement push(), isEmpty(), peek() and size() for stack

### LINKED ISSUE #27: Implement `push(element: E)` function for Stacks
Taking reference from issue [#14](https://github.com/code-sherpas/kollections/issues/14), implement the `push(element: E)`  function for a Stack. This function adds an element to the top of the stack.

## First step

Search, read, and understand information about stacks and how they are implemented.

## Second step

After you open a pull request with the data structure implementation, wait for comments or approval from @code-sherpas/kollections-maintainers.

## Minimum acceptance criteria

- The implementation must be well documented with proper [Javadoc](https://es.wikipedia.org/wiki/Javadoc) at Class and function levels.
- The implementation must be tested with automatic tests that cover all relevant cases and situations.
- The implementation must be [generic](https://kotlinlang.org/docs/generics.html).
- The implementation must have the expected time complexity according to Computer Science. [Here you can find a very helpful cheat sheet](https://www.bigocheatsheet.com/).
- Compare efficiency using a dynamic array as underlying data structure vs using a singly linked list. Choose the most efficient.

## Related resources:

- https://www.youtube.com/watch?v=v4cd1O4zkGw&ab_channel=HackerRank
- https://www.youtube.com/watch?v=wjI1WNcIntg&ab_channel=HackerRank
- https://en.wikipedia.org/wiki/Stack_(abstract_data_type)
- https://www.bigocheatsheet.com/

### PR DESCRIPTION
**Pull request by thebongy**
Closes #27, #29, #30, #31

### CODE REVIEW

File: lib/src/test/kotlin/main/StackTests.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class StackTests {
+    @Test
+    fun checkPushAddsElements() {
+        val stack = Stack<Int>()
+        assertThat(stack.size()).isEqualTo(0)
+        assertThat(stack.isEmpty()).isTrue
+
+        for (i in 1..10) {
```
Comment (ashikka) [inline lib/src/test/kotlin/main/StackTests.kt:17]: @thebongy Could you write another test with another data type? Preferably, a custom data type?

**Review by ashikka — CHANGES_REQUESTED** [review]

File: lib/src/test/kotlin/main/StackTests.kt (hunk 1 above)
Comment (thebongy) [reply lib/src/test/kotlin/main/StackTests.kt:17]: On it @ashikka

Comment (ashikka) [reply lib/src/test/kotlin/main/StackTests.kt:17]: Morever, @thebongy please do not PR to the `main` branch. Kindly PR to the respective data structure branches.

Comment (thebongy) [reply lib/src/test/kotlin/main/StackTests.kt:17]: My bad, I have switched to the stack branch

Comment (thebongy) [reply lib/src/test/kotlin/main/StackTests.kt:20]: @ashikka I have added a test with a custom data type, can you please review it?

Comment (ashikka) [reply lib/src/test/kotlin/main/StackTests.kt:20]: Great job! @thebongy :sparkling_heart:

**Review by ashikka — APPROVED** [review]

**Pull request merged**